src/tests/messageUtils.ts

```diff
@@ -0,0 +1,280 @@
+   * This is not recommended, use for demo purposes only!
+   *
+   * @param lightDid The light DID whose keys will be used on the full DID.
+   * @returns A full DID instance that is not yet written to the blockchain.
+   */
+export async function createLocalDemoFullDidFromLightDid(
+  lightDid: DidDocument
+): Promise<DidDocument> {
+  const { uri, authentication } = lightDid
+
+  return {
+    uri: Did.getFullDidUri(uri),
+    authentication,
+    assertionMethod: authentication,
+    capabilityDelegation: authentication,
+    keyAgreement: lightDid.keyAgreement,
+  }
+}
+
+
+/**
+   * Generates a callback that can be used for decryption.
+   *
+   * @param secretKey The options parameter.
+   * @param secretKey.secretKey The key to use for decryption.
+   * @returns The callback.
+   */
+export function makeDecryptCallback({
+  secretKey,
+}: KiltEncryptionKeypair): DecryptCallback {
+  return async function decryptCallback({ data, nonce, peerPublicKey }) {
+    const decrypted = Crypto.decryptAsymmetric(
+      { box: data, nonce },
+      peerPublicKey,
+      secretKey
+    )
+    if (decrypted === false) throw new Error('Decryption failed')
+    return { data: decrypted }
+  }
+}
+
+export type EncryptionKeyToolCallback = (
+    didDocument: DidDocument
+  ) => EncryptCallback
+
+
+
+/**
+   * Generates a callback that can be used for encryption.
+   *
+   * @param secretKey The options parameter.
+   * @param secretKey.secretKey The key to use for encryption.
+   * @returns The callback.
+   */
+export function makeEncryptCallback({
+  secretKey,
+}: KiltEncryptionKeypair): EncryptionKeyToolCallback {
+  return (didDocument) => {
+    return async function encryptCallback({ data, peerPublicKey }) {
+      const keyId = didDocument.keyAgreement?.[0].id
+      if (!keyId) {
+        throw new Error(`Encryption key not found in did "${didDocument.uri}"`)
+      }
+      const { box, nonce } = Crypto.encryptAsymmetric(
+        data,
+        peerPublicKey,
+        secretKey
+      )
+      return {
+        nonce,
+        data: box,
+        keyUri: `${didDocument.uri}${keyId}`,
+      }
+    }
+  }
+}
+
+export interface EncryptionKeyTool {
+    keyAgreement: [KiltEncryptionKeypair]
+    encrypt: EncryptionKeyToolCallback
+    decrypt: DecryptCallback
+  }
+
+/**
+   * Generates a keypair suitable for encryption.
+   *
+   * @param seed {string} Input to generate the keypair from.
+   * @returns Object with secret and public key and the key type.
+   */
+export function makeEncryptionKeyTool(seed: string): EncryptionKeyTool {
+  const keypair = Crypto.makeEncryptionKeypairFromSeed(blake2AsU8a(seed, 256))
+
+  const encrypt = makeEncryptCallback(keypair)
+  const decrypt = makeDecryptCallback(keypair)
+
+  return {
+    keyAgreement: [keypair],
+    encrypt,
+    decrypt,
+  }
+}
+
+// Mock function to generate a key ID without having to rely on a real chain metadata.
```

Comment:
Is this comment up to date? Mock should only be used for tests?

Reply:
AFAIK, this comment is still up-to-date. The function is only used inside the `messageUtils` file.

Reply:
ah i didn'T see that his is in the `test/` directory